test/jdk/java/lang/StackWalker/SanityTest.java

```diff
@@ -25,78 +25,98 @@
+        assertThrows(NullPointerException.class, () ->
+                StackWalker.getInstance().walk(null));
     }
 
     @Test
-    public static void testNullConsumer() {
-        try {
-            StackWalker.getInstance().forEach(null);
-            throw new RuntimeException("NPE expected");
-        } catch (NullPointerException e) {}
+    public void testNullConsumer() {
+        assertThrows(NullPointerException.class, () ->
+                StackWalker.getInstance().forEach(null));
+    }
+
+    @ParameterizedTest
+    @MethodSource("noRetainClassRef")
+    public void testUOEFromGetDeclaringClass(StackWalker sw) {
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::getDeclaringClass));
     }
 
+    @ParameterizedTest
+    @MethodSource("noRetainClassRef")
+    public void testUOEFromGetMethodType(StackWalker sw) {
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::getMethodType));
+    }
 
-    @Test
-    public static void testUOEFromGetDeclaringClass() {
-        try {
-            StackWalker sw = StackWalker.getInstance();
-            sw.forEach(StackWalker.StackFrame::getDeclaringClass);
-            throw new RuntimeException("UOE expected");
-        } catch (UnsupportedOperationException expected) {
-        }
+    private static Stream<StackWalker> noMethodInfo() {
+        return Stream.of(StackWalker.getInstance(CLASS_INFO),
+                         StackWalker.getInstance(CLASS_INFO, RETAIN_CLASS_REFERENCE));
     }
 
-    @Test
-    public static void testUOEFromGetMethodType() {
-        try {
-            StackWalker sw = StackWalker.getInstance();
-            sw.forEach(StackWalker.StackFrame::getMethodType);
-            throw new RuntimeException("UOE expected");
-        } catch (UnsupportedOperationException expected) {}
+    @ParameterizedTest
+    @MethodSource("noMethodInfo")
+    public void testNoMethodInfo(StackWalker sw) {
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::getMethodName));
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::getMethodType));
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::getDescriptor));
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::getByteCodeIndex));
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::getFileName));
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::isNativeMethod));
+        assertThrows(UnsupportedOperationException.class, () ->
+                sw.forEach(StackWalker.StackFrame::toStackTraceElement));
```

Comment:
Should we check that the exception is thrown by each frame? This code will hit the exception at the first frame and not check the others.

Reply:
I think not necessary since the walker returns the `ClassFrameInfo` for all frames.